File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[] A = Arrays.stream(br.readLine().split(" "))
                .mapToInt(Integer::parseInt)
                .toArray();

        int M = Integer.parseInt(br.readLine());
        int[] B = Arrays.stream(br.readLine().split(" "))
                .mapToInt(Integer::parseInt)
                .toArray();

        solution(A, B);
    }

    public static void solution(int[] A, int[] B) {
        Arrays.sort(A);

        for (int value : B) {
            System.out.println(binarySearch(value, A));
        }
    }

    public static int binarySearch(int value, int[] arr) {
        int l = 0;
        int r = arr.length - 1;

        while (l <= r) {
            int mid = (l + r) / 2;

            if (arr[mid] < value) {
                l = mid + 1;
            }
            else if (arr[mid] > value) {
                r = mid - 1;
            }
            else return 1;
        }
        return 0;
    }
}

File: 백준/Silver/1920. 수 찾기/수 찾기.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()
    val arrA = StringTokenizer(br.readLine()).run {
        IntArray(countTokens()) { nextToken().toInt() }
    }
    val m = br.readLine().toInt()
    val arrB = StringTokenizer(br.readLine()).run {
        IntArray(countTokens()) { nextToken().toInt() }
    }

    arrA.sort()
    arrB.forEach {
        if (isExist(it, arrA)) {
            println(1)
        } else {
            println(0)
        }
    }
}

private fun isExist(x: Int, arr: IntArray): Boolean {
    var l = 0
    var r = arr.size - 1

    while(l <= r) {
        val mid = (l + r) / 2

        if (x > arr[mid]) {
            l = mid + 1
        } else if (x < arr[mid]) {
            r = mid - 1
        } else {
            return true
        }
    }

    return false
}